python/core/memory_monitor.py

```
@@ -0,0 +1,259 @@
+                    elif usage_ratio >= self.warning_threshold:
+                        self._warning_count += 1
+                        print(f"[MEMORY] ⚠️  WARNING: {snapshot}")
+                    else:
+                        # Log periodically even when healthy (every 10 minutes)
+                        if len(self._snapshots) % 10 == 0:
+                            print(f"[MEMORY] ✅ Healthy: {snapshot}")
+
+                except asyncio.CancelledError:
+                    print("[MEMORY] Monitor task cancelled")
+                    break
+                except Exception as e:
```

Ruff (0.14.2)

Do not catch blind exception: `Exception`

(BLE001)